File: JavaScript/6kyu/Longest alphabetical substring.js
function longest(str) {
  let alphabet = `abcdefghijklmnopqrstuvwxyz`
  let temp = ``
  let maxSubstring = ``
  
  for(let i = 0; i <= str.length; ++i){
    if(alphabet.indexOf(str[i]) <= alphabet.indexOf(str[i+1])){
      temp += str[i]
    }else{
      temp += str[i]
      maxSubstring = maxSubstring.length >= temp.length ? maxSubstring : temp
      temp = ``
    }
  }
      
  return maxSubstring
}

File: Python/6kyu/Longest alphabetical substring.py
def longest(s):
    if len(s) <= 1:
        return s
    
    alphabet = list("abcdefghijklmnopqrstuvwxyz")
    temp = ""
    maxStr = ""
    
    for idx, x in enumerate(s):
        if idx + 1 == len(s):
            if(len(temp) > 0):
                if(alphabet.index(temp[-1]) <= alphabet.index(x)):
                    temp = temp + x
                if len(maxStr) < len(temp):
                    maxStr = temp
            break
        if alphabet.index(x) <= alphabet.index(s[idx+1]):
            temp = temp + x
        else:
            temp = temp + x
            if len(maxStr) < len(temp):
                maxStr = temp
            temp = ""
        
    return maxStr
